### ISSUE TITLE
Silently skipping struct members in storage-memory copies might lead to confusing results.

### ISSUE THREAD

**Issue by ekpyron** [ethereum/solidity#6444]
Consider the following contract:
```
contract C {
    struct S {
        mapping(uint=>uint) m;
    }
    S s1;
    S s2;
    function f() public returns (uint256) {
        s1.m[0] = 23;
        S memory tmp = s1;
        // accessing tmp.m will result in an error, but we can still assign back to a storage struct
        s2 = tmp;
        return s2.m[0];
    }
}
```
It compiles without warning and returns ``0``, although it very much looks like it should return ``23``.

The reason is that ``mapping``s cannot "live outside storage" and are thus silently ignored while copying the struct. Accessing ``mapping``s in memory structs causes an error. However, assigning back to a storage struct works and again silently skips the ``mapping``. IMO this results in counter-intuitive behavior.

What are the benefits of allowing to assign storage structs with members that can't live outside storage to memory structs in general?

**Comment by ekpyron** [comment]
Current tendency seems to be disallowing mappings (resp. types that can't live outside storage) in memory structs altogether. The subset of the struct that is copied now could still be manually copied member by member instead.

**Comment by ekpyron** [comment]
Related to or might even be fixed by #6282.

**Comment by chriseth** [comment]
Closed by https://github.com/ethereum/solidity/pull/9146
